src/parse_instruction.py

```diff
@@ -283,6 +262,14 @@ def constant_fold(arg: Argument) -> Argument:
     return arg
 
 
+def get_jump_target(label: Argument) -> JumpTarget:
+    if isinstance(label, AsmGlobalSymbol):
+        return JumpTarget(label.symbol_name)
+    if not isinstance(label, JumpTarget):
+        raise DecompFailure(f"Couldn't parse instruction: invalid branch target")
```

Comment:
(a bit inconsistent to use an exception here but asserts elsewhere in Arch.parse(). Doesn't matter much either way though since we suppress exception messages in parse_instruction)

Reply:
That's fair - shifting this around means the exception is now suppressed.